control-plane/agents/src/bin/core/controller/scheduling/volume_policy/pool.rs

```diff
@@ -34,6 +37,15 @@ impl PoolBaseFilters {
             false => item.pool.free_space() > request.size,
         }
     }
+    /// Return true if the pool has enough capacity to resize the replica by requested
+    /// value.
+    pub(crate) fn min_free_space_repl_resize(
```

Comment:
I think we should modify min_free_space to take into account resize, otherwise seems like we're doing checks for commitment here?